src/libraries/System.Private.CoreLib/src/System/Number.Parsing.cs

```diff
@@ -358,7 +358,7 @@ private static unsafe bool TryParseNumber(ref char* str, char* strEnd, NumberSty
                         {
                             number.Scale++;
                         }
-                        else if (digCount < maxDigCount)
+                        else if (digCount < maxDigCount && number.Kind == NumberBufferKind.FloatingPoint)
```

Comment:
I wonder if we can improve this a bit. The additional `number.Kind == NumberBufferKind.FloatingPoint` won't be elided but it really should be.

Reply:
Fixed